chatsky/pipeline/pipeline/component.py

```diff
@@ -134,9 +150,35 @@ async def run_extra_handler(self, stage: ExtraHandlerType, ctx: Context, pipelin
             logger.warning(f"{type(self).__name__} '{self.name}' {extra_handler.stage} extra handler timed out!")
 
     @abc.abstractmethod
-    async def run_component(self, ctx: Context, pipeline: Pipeline) -> None:
+    async def run_component(self, ctx: Context, pipeline: Pipeline) -> Optional[ComponentExecutionState]:
+        """
+        Method for running this component. It can be an Actor, Service or ServiceGroup.
+        It has to be defined in the child classes,
+        which is done in each of the default PipelineComponents.
+        Service 'handler' has three possible signatures. These possible signatures are:
+
+        - (ctx: Context) - accepts current dialog context only.
+        - (ctx: Context, pipeline: Pipeline) - accepts context and current pipeline.
+        - | (ctx: Context, pipeline: Pipeline, info: ServiceRuntimeInfo) - accepts context,
+              pipeline and service runtime info dictionary.
+
+        :param ctx: Current dialog context.
+        :param pipeline: The current pipeline.
+        :return: `None`
+        """
         raise NotImplementedError
 
+    @property
+    def computed_name(self) -> str:
+        """
+        Every derivative of `PipelineComponent` must define this property.
+        :return: `str`.
```

Comment:
```suggestion
        Default name that is used if `self.name` is not defined.
        In case two components in a `ServiceGroup` have the same `computed_name`
        an incrementing number is appended to the name.
```
Also, link the entities in the doc to their doc entries. You can link to `self.name` with 
```
:py:attr:`~.PipelineComponent.name`
```
The breakdown of the syntax:

- `~` is used to leave only the last component displayed (in markdown terms it's `[name](link)` instead of `[PipelineComponent.name](link)`). TBH this isn't necessary here; mostly used if a non-local object is referenced.
- `.` is used to reference a local object (`.Service` tells to find a `Service` entity in the current file; otherwise have to specify full path `chatsky.pipeline.service.service.Service`)
- `:py:attr:` specifies the type of the object referenced. Available types can be found here: https://www.sphinx-doc.org/en/master/usage/domains/python.html#cross-referencing-python-objects